hyperspy/_components/expression.py

```diff
@@ -215,3 +218,13 @@ def compile_function(self, module="numpy", position=False):
                         self,
                         Expression)
                     )
+
+def check_expression_is_linear(expression, name):
+    if name not in expression:
+        raise AttributeError("The parameter '" + name + "' is not in the expression.")
+    try:
+        if not sympy.Eq(sympy.diff(expression, name, 2), 0):
+            return False
+    except TypeError:
+        return False
+    return True
```

Comment:
I still think this needs a rewrite in the form I suggested previously (while obviously keeping the first check that the parameter is indeed in the expression).

Currently this reads:
```python
try:
    if not True:
        return False
    else:
        return True
except:
    return False
```

Which is clearly simpler this way:
```python
try:
    return [True/False] # the actual bool statement here. Add "not" if needed, which I don't think you do
except:
    return False
```